Documentation/Storage-Configuration/Object-Storage-RGW/object-storage.md

```diff
@@ -243,6 +243,135 @@ s5cmd --endpoint-url http://$AWS_HOST:$PORT cp s3://$BUCKET_NAME/rookObj /tmp/ro
 cat /tmp/rookObj-download
 ```
 
+### Create a Bucket via Ceph COSI Driver
+
+The Ceph COSI driver will be started automatically with default settings when first Object Store gets created. The driver will be deleted when Rook Operator is uninstalled. The driver will be deployed in the same namespace as Rook operator. The COSI controller needs already running if the Ceph COSI Driver needs to come up. The COSI controller can be deployed by following the commands below:
+
+```bash
+kubectl apply -k github.com/kubernetes-sigs/container-object-storage-interface-api ## installs CRDs
+kubectl apply -k github.com/kubernetes-sigs/container-object-storage-interface-controller ## installs COSI controller
+```
+
+Now user can define BucketClass and BucketClaim to create a bucket. The BucketClass and BucketClaim can be defined as below:
+
+```yaml
+kind: BucketClass
+apiVersion: objectstorage.k8s.io/v1alpha1
+metadata:
+  name: sample-bcc
+driverName: ceph.objectstorage.k8s.io
+deletionPolicy: Delete
+parameters:
+  ObjectStoreUserSecretName: rook-ceph-object-user-<store_name>-cosi
+  ObjectStoreNamespace: rook-ceph
+---
+kind: BucketClaim
+apiVersion: objectstorage.k8s.io/v1alpha1
+metadata:
+  name: sample-bc
+spec:
+  bucketClassName: sample-bcc
+  protocol
+```
+
+The `CephObjectStoreUser` named cosi will be automatically and user can refer to the secret name in the BucketClass. The BucketClass and BucketClaim can be created by using above the yaml file. After successful creation of above CRDs user can fetch them by using below commands:
```

Comment:
Agreed that we don't expect Rook to create any CRs, since we expect the users to create them. Having Rook internally create the user and the secret makes sense. This can be implemented by the object store calling the object store user controller code for creating the built-in user and secret. This way we get the benefit of sharing code, but we don't need a CR to trigger it, or worry about the user deleting the CR.

Reply:
Will add the automatic creation as Update on the Initial PR